joomla/joomla-cms
# [4.0] Save sidebar state using localStorage

Pull Request for Issue #15892 

### Summary of Changes

This PR adds the use of `localStorage` to save the state of the sidebar (open/closed). Upon page refresh it wil automatically open or close the sidebar based on the state it was in before.

### Testing Instructions

**1.**
- Apply patch (sidebar should be open)

**2.** 
- Close the sidebar
- Refresh the page
- Sidebar should stay closed

**3.** 
- Open the sidebar
- Refresh the page
- Sidebar should stay open

@dgt41 I've added the test to see if `localStorage` can be used in the `core.js`, so you can use this from now on. Just do `if (Joomla.localStorageEnabled()) { ... }`
